Fightdet/data_generator.py

```diff
@@ -0,0 +1,143 @@
+        for i,label in enumerate(self.dirs):
+            print('%10s : '%(label),i)
+        return None
+
+    def __len__(self):
+        # calculate the iterations of each epoch
+        steps_per_epoch = np.ceil(len(self.X_path) / float(self.batch_size))
+        return int(steps_per_epoch)
+
+    def __getitem__(self, index):
+        """
+        Get the data of each batch
+        """
+        # get the indexs of each batch
+        batch_indexs = self.indexes[index*self.batch_size:(index+1)*self.batch_size]
+        # using batch_indexs to get path of current batch
+        batch_path = [self.X_path[k] for k in batch_indexs]
+        # get batch data
+        batch_x, batch_y = self.data_generation(batch_path)
+        return batch_x, batch_y
+
+    def on_epoch_end(self):
+        # shuffle the data at each end of epoch(if shuffle=True)
+        if self.shuffle == True:
+            np.random.shuffle(self.indexes)
+
+    def data_generation(self, batch_path):
+        # load data into memory
+        batch_x = [self.load_data(x) for x in batch_path]
+        batch_y = [self.Y_dict[x] for x in batch_path]
+        # transfer the data format
+
+        # shape of batch x = (batch_size,64,224,224,3)
+        batch_x = np.array(batch_x)
+        # shape of batch y = (batch_size)
+        batch_y = np.array(batch_y)
+        return batch_x, batch_y
+
+    def standardize(self, data):
+        mean = np.mean(data)
+        std = np.std(data)
+        return (data-mean) / std
+
+    def random_flip(self, video, prob):
+        s = np.random.rand()
+        if s < prob:
+          #flipping frame in a video horizontally
+          video = np.flip(m=video, axis=2)
+        return video
+
+    def load_data(self, path):
+        # load the processed .npy files which have 3 channels (1 for gray_frame, 4-5 for optical flows)
+        data = np.load(path, mmap_mode='r')
+        data = np.float32(data)
+        # whether to utilize the data augmentation
+        if self.data_aug:
+            data = self.random_flip(data, prob=0.5)
+        # standardize gray images and optical flows, respectively
+        data[...,:1] = self.standardize(data[...,:1])
+        data[...,1:] = self.standardize(data[...,1:])
+        return data
+
+# Instatiate Generator
+def instantiate_generator(batch_size=5):
+    '''
+    This function instantiate the data generators for train and validation set for CNN model training
+    Parameters:
+        batch size(int): batch size for model training, default is 5, take your
+        processing memory into account if you wish to change it. Model might not fit
+        if the system memory is insufficient.
+
+    Returns:
+        train_generator(Datagenerator): data generator class for train set
+        val_generator(Datagenerator): data generator class for validation set
+
+    '''
+    # change the folder path to raw_data folder which contain train and test set
+    npy_dataset_folder = 'raw_data/npy_raw_data'
```

Comment:
`npy_dataset_folder` should be a parameter for this function since `raw_data` is not in this repo and we may store raw_data folder at a different file path.

Reply:
yeah It'll be better.